single_kernel_mongo/events/backups.py

```diff
@@ -119,14 +122,43 @@ def _on_s3_credential_changed(self, event: CredentialsChangedEvent) -> None:  #
             )
             return
 
+        if not self.manager.validate_s3_config():
+            logger.warning(
+                "Relation to S3 charm exists but not all necessary configurations have been set."
+            )
+            self.manager.state.statuses.set(
+                BackupStatuses.PBM_MISSING_CONFIGS.value,
+                scope="unit",
+                component=self.manager.name,
+            )
+            return
+
         # Get the credentials from S3 connection
         credentials = self.s3_client.get_s3_connection_info()
 
         try:
+            # First create the bucket if it does not exist.
+            self.manager.create_bucket(credentials=credentials)
+            # Then set the config options on PBM.
             self.manager.set_config_options(credentials=credentials)
+            # Finally, resync the configuration.
             self.manager.resync_config_options()
+        except InvalidS3CredentialsError:
+            logger.error("Incorrect S3 credentials")
+            self.manager.state.statuses.add(
+                BackupStatuses.PBM_INCORRECT_CREDS.value, scope="unit", component=self.manager.name
+            )
+            return
+        except (FailedToCreateS3BucketError, SSLError, ConnectTimeoutError) as err:
+            logger.error("Failed to create bucket: %s", err)
+            self.manager.state.statuses.add(
+                BackupStatuses.FAILED_TO_CREATE_BUCKET.value,
+                scope="unit",
+                component=self.manager.name,
+            )
+            event.defer()
+            return
         except SetPBMConfigError:
-            logger.error("Failed to configure s3 backup options")
```

Comment:
why not keep?

Reply:
It's already logged when the error is raised with more information. Prevent duplication of logs.